lib/stat_tracker.rb

```diff
@@ -41,202 +41,211 @@ def count_of_teams
     end
-    games_won_array
-  end
 
-  def games_won_into_hash_helper
-  games_won_hash = total_games_won_by_coach_array_helper.group_by(&:itself)
-    games_won_hash.map{|key, value| [key, value.length]}.to_h
-  end
+    # Creates hash with 'head_coach' as key and value as an array with first
+    # element as integer of games won and second element integer of all other games (losses or ties)
+    season_games = season_games.group_by do |game|
+      game["head_coach"]
+    end.transform_values! {|games| games.partition {|game| game["result"] == "WIN"}.map(&:length)}
 
-  def coaches_with_games_played_and_won_array_helper
-      coaches_with_games_played_array = total_games_played_by_coach_helper.keys
-      coaches_with_games_won_array = games_won_into_hash_helper.keys
-      coaches_with_games_won_and_played_array = coaches_with_games_played_array & coaches_with_games_won_array
-      coaches_with_games_won_and_played_array
+    # Returns name of coach with greatest win percentage (games won divided by total of games played)
+    season_games.max_by do |coach, games|
+      games[0].to_f / games.sum
+    end.first
   end
 
-  def coaches_winning_percentage_hash_helper
-    coaches_winning_percentage_hash = {}
-    coaches_with_games_played_and_won_array_helper.each do |coach|
+  def worst_coach(season_id)
+    # Find all games in season & return array of those games' ids
+    season_games = @games.find_all do |game|
+      game["season"] == season_id
+    end.map {|game| game["game_id"]}
 
-      coaches_winning_percentage_hash[coach] = ((games_won_into_hash_helper[coach] / total_games_played_by_coach_helper[coach] * 100))
+    # Find all those games in @game_teams & return array of CSV row objects
+    season_games = @game_teams.find_all do |game|
+      season_games.include?(game["game_id"])
     end
-    coaches_winning_percentage_hash
+
+    # Creates hash with 'head_coach' as key and value as an array with first
+    # element as integer of games won and second element integer of all other games (losses or ties)
+    season_games = season_games.group_by do |game|
+      game["head_coach"]
+    end.transform_values! {|games| games.partition {|game| game["result"] == "WIN"}.map(&:length)}
+
+    # Returns name of coach with greatest win percentage (games won divided by total of games played)
+    season_games.min_by do |coach, games|
+      games[0].to_f / games.sum
+    end.first
   end
 
-  def winningest_coach
-    name_of_coach_with_number_variable = coaches_winning_percentage_hash_helper.find_all do |key, value|
-      value == coaches_winning_percentage_hash_helper.values.max
-    end
-    name_of_coach_array_variable = []
-  name_of_coach_with_number_variable.each do |name_and_number|
-  name_of_coach_array_variable << name_and_number[0]
-    end
-    name_of_coach_array_variable.join(", ")
+  def most_accurate_team(season_id)
+    # Find all games in season & return array of those games' ids"
```

Comment:
Nice to see this chunk of code at work again in a different method! I think this will serve us well going forward.